src/sage/rings/padics/witt_vector_ring.py

```diff
@@ -263,27 +262,33 @@ def __classcall_private__(cls, coefficient_ring, prec=1, p=None, algorithm=None)
 
         return child.__classcall__(child, coefficient_ring, prec, p)
 
-    def __init__(self, coefficient_ring, prec, prime):
+    def __init__(self, coefficient_ring, prec, prime) -> None:
         r"""
-        Initialises ``self``.
+        Initialise ``self``.
 
         EXAMPLES::
 
-            sage: W = WittVectorRing(PolynomialRing(GF(5), 't'), prec=4)
-            sage: W
+            sage: W = WittVectorRing(PolynomialRing(GF(5), 't'), prec=4); W
             Ring of truncated 5-typical Witt vectors of length 4 over Univariate Polynomial Ring in t over Finite Field of size 5
             sage: type(W)
             <class 'sage.rings.padics.witt_vector_ring.WittVectorRing_phantom_with_category'>
 
             sage: TestSuite(W).run()
         """
-        self._coefficient_ring = coefficient_ring
+        cring = coefficient_ring
+        self._coefficient_ring = cring
         self._prec = prec
         self._prime = prime
 
-        CommutativeRing.__init__(self, self)
+        if (cring.characteristic() > 0 and prec == 1
+                and cring in IntegralDomains()):
```

Comment:
```suggestion
        if cring in IntegralDomains() and prec == 1:
```
$W_1(\mathbb Q)$ is fine though.